### PR TITLE
Documenting the changes in v2.4.2

### PR DESCRIPTION
**Pull request by ahadas**
This PR documents all [the changes that are introduced in v2.4.2](https://issues.redhat.com/projects/MTV/versions/12410052) and need to be documented

### CODE REVIEW

**Comment by anarnold97** [comment]
@ahadas, thanks will do myself or ask @RichardHoch  to do the review (if that is ok)

@RichardHoch - do you have time to pick this up? if not i will get to it tomorrow evening

File: documentation/modules/creating-vddk-image.adoc [hunk 1]
```diff
@@ -30,8 +30,14 @@ Storing the VDDK image in a public registry might violate the VMware license ter
...
-. In a browser, navigate to the link:https://developer.vmware.com/web/sdk/7.0/vddk[VMware VDDK version 7 download page].
-. Select version 7.0.3.2 and click *Download*.
+. In a browser, navigate to the link:https://developer.vmware.com/web/sdk/8.0/vddk[VMware VDDK version 8 download page].
+. Select version 8.0.1 and click *Download*.
+
+[NOTE]
+====
+In order to migrate to {virt} 4.12 (or lower), version 7.0.3.2 of VDDK should be downloaded from link:https://developer.vmware.com/web/sdk/7.0/vddk[VMware VDDK version 7 download page] instead.
+====
+
```
Comment (RichardHoch) [inline documentation/modules/creating-vddk-image.adoc:40]: ... 4.12 or earlier, download VDDK version 7.0.3. from link:https://developer.vmware.com/web/sdk/7.0/vddk[VMware VDDK version 7 download page].

File: documentation/modules/rn-2.4.adoc [hunk 2]
```diff
@@ -146,3 +150,11 @@ Previously, the `cluster-admin` role was required to browse and create providers
 .Convert i440fx to q35
 
 Migration of virtual machines with i440fx chipset is now supported. The chipset is converted to q35 during the migration. link:https://issues.redhat.com/browse/MTV-430[(MTV-430)]
+
+.Preserve the UUID setting in SMBIOS for a VM that is migrated from RHV
+
+The Universal Unique ID (UUID) number within the System Management BIOS (SMBIOS) doesn't change anymore for VMs that are migrated from RHV. This fix enables applications that run within the guest operating system and depend on this setting (e.g., for licensing) to operate on the target OCP cluster the same way as they did on RHV. link:https://issues.redhat.com/browse/MTV-597[(MTV-597)]
+
```
Comment (RichardHoch) [inline documentation/modules/rn-2.4.adoc:157]: "... (SMBIOS) no longer changes for VMs that are migrated from RHV. This enhancement enables applications that run within the guest operating system and depend on this setting (for example, for licensing) to operate on the target OCP cluster the same way as they did on RHV. link:https://issues.redhat.com/browse/MTV-597[(MTV-597)]."

File: documentation/modules/rn-2.4.adoc [hunk 3]
```diff
@@ -34,6 +34,10 @@ The migrated virtual machines are no longer scheduled on the target OpenShift Co
 
 You must update the `StorageProfile` resource with `accessModes` and `volumeMode` for non-provisioner storage classes such as NFS.
 
+.VDDK 8 can be used in the VDDK image
+
+Previously, version 7 of VMware Virtual Disk Development Kit (VDDK) should have been used when creating the VDDK image. In this release, the VDDK image can also be created with version 8 of VDDK. VDDK 8 is recommended when migrating to OCP 4.13 or later. In order to migrate to OCP 4.12 or lower, VDDK 7 should be used.
+
```
Comment (RichardHoch) [inline documentation/modules/rn-2.4.adoc:40]: Avoid the word "should."
Previous versions of {project-short} supported only using VDDK version 7 for the VDDK image.  {project-short} supports both versions 7 and 8 as follows: 
* If you are migrating to OCP 4.12 or earlier, use VDDK version 7.
* If you are migrating to OCP 4.13 or later, use VDDK version 8.

use the attribute for OCP.

File: documentation/modules/rn-2.4.adoc [hunk 4]
```diff
@@ -146,3 +150,11 @@ Previously, the `cluster-admin` role was required to browse and create providers
...
 
 Migration of virtual machines with i440fx chipset is now supported. The chipset is converted to q35 during the migration. link:https://issues.redhat.com/browse/MTV-430[(MTV-430)]
+
+.Preserve the UUID setting in SMBIOS for a VM that is migrated from RHV
+
+The Universal Unique ID (UUID) number within the System Management BIOS (SMBIOS) doesn't change anymore for VMs that are migrated from RHV. This fix enables applications that run within the guest operating system and depend on this setting (e.g., for licensing) to operate on the target OCP cluster the same way as they did on RHV. link:https://issues.redhat.com/browse/MTV-597[(MTV-597)]
+
+.Do not expose password for RHV in error messages
+
+Previously, the password that was specified for ovirt-engine appeared in error messages that were displayed in the web console and logs when faling to connect to RHV. In this release, error messages that are generated when failing to connect to RHV do not reveal the password for ovirt-engine.
```
Comment (RichardHoch) [inline documentation/modules/rn-2.4.adoc:160]: Is "ovirt-engine" an attribute or the same for upstream and downstream? 
"...that were displayed in the web console and logs when MTV failed to connect to RHV. In this release, error messages that are generated when MTV fails to connect to RHV do not reveal the password for ovirt-engine. "

Use attributes for MTV and RHV.

**Comment by RichardHoch** [comment]
@anarnold97 I made some suggestions -- please give them a lookover.

**Comment by RichardHoch** [comment]
@ahadas In documentation, we try to avoid the word "should" because it can either mean "must" or "it's a good idea." I used to teach English in Israel and I was surprised to see Israelis always translate צריך as "should" instead of "must."

File: documentation/modules/rn-2.4.adoc (hunk 4 above)
Comment (ahadas) [reply documentation/modules/rn-2.4.adoc:160]: > @ahadas Is "ovirt-engine" an attribute or the same for upstream and downstream?

well, that's a good point - the underlying component is the same (ovirt-engine) but we better call it RHVM on downstream here

**Comment by ahadas** [comment]
> @ahadas In documentation, we try to avoid the word "should" because it can either mean "must" or "it's a good idea." I used to teach English in Israel and I was surprised to see Israelis always translate צריך as "should" instead of "must."

in our defense, it's not just us - google translate also translates it this way (which doesn't make it correct of course) :) 

and sure, I posted this content and expected your follow up to polish it for the documentation
@RichardHoch your comments above make sense to me, I replied on one of the them to provide more information - could you please make these changes once you reach an agreement with Andy?

File: documentation/modules/creating-vddk-image.adoc [hunk 5]
```diff
@@ -30,8 +30,14 @@ Storing the VDDK image in a public registry might violate the VMware license ter
 $ mkdir /tmp/<dir_name> && cd /tmp/<dir_name>
 ----
 
-. In a browser, navigate to the link:https://developer.vmware.com/web/sdk/7.0/vddk[VMware VDDK version 7 download page].
-. Select version 7.0.3.2 and click *Download*.
+. In a browser, navigate to the link:https://developer.vmware.com/web/sdk/8.0/vddk[VMware VDDK version 8 download page].
+. Select version 8.0.1 and click *Download*.
```
Comment (ahadas) [inline documentation/modules/creating-vddk-image.adoc:34]: @qiyuann please double check that this is the version that you're going to test with

File: documentation/modules/rn-2.4.adoc (hunk 3 above)
Comment (anarnold97) [reply documentation/modules/rn-2.4.adoc:43]: changed

File: documentation/modules/rn-2.4.adoc (hunk 2 above)
Comment (anarnold97) [reply documentation/modules/rn-2.4.adoc:160]: changed

File: documentation/modules/rn-2.4.adoc (hunk 4 above)
Comment (ahadas) [reply documentation/modules/rn-2.4.adoc:163]: commented on #398

**Comment by ahadas** [comment]
closed in favor of #398

**Pull request closed without merging**